share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -353,6 +363,93 @@ function(ExternalZephyrProject_Cmake)
   import_kconfig(CONFIG_ ${BINARY_DIR}/zephyr/.config TARGET ${ZCMAKE_APPLICATION})
 endfunction()
 
+# Usage:
+#   ExternalZephyrProject_Group(<name>
+#                               [INCLUDE <images>]
+#                               [EXCLUDE <images>]
+#                               [INCLUDE_FROM <groups>]
+#                               [EXCLUDE_FROM <groups>]
+#                               [IMMUTABLE]
+#   )
+#
+# This function defines groups of external Zephyr projects. Images added using
+# `ExternalZephyrProject_Add()` may inherit common properties of the group(s)
+# they are assigned to.
+#
+# This function can either add a new group "<name>", or extend the definition of
+# an existing group with that name. This lets multiple sources influence what a
+# given group should represent in the multiimage build system. The final set of
+# images/properties for every group is evaluated in `sysbuild_groups_resolve()`.
+#
+# <name>                 Name of the group.
+# INCLUDE <images>:      Add a set of images to the group.
+# EXCLUDE <images>:      Remove a set of images from the group.
+# INCLUDE_FROM <groups>: INCLUDE all images belonging to each group in the list.
+# EXCLUDE_FROM <groups>: EXCLUDE all images belonging to each group in the list.
+# IMMUTABLE:             Disallow extending the definition of this group.
```

Comment:
At current stage I don't see any reasons why sysbuild should support this.

It will just result in users having to look into the sysbuild CMake code to understand how to manipulate the right properties, so atm it seems just providing false impression that a group is locked.

Reply:
Continued: https://github.com/zephyrproject-rtos/zephyr/pull/58174#discussion_r1201662733